### PR TITLE
Travis build stages

### PR DESCRIPTION
**Pull request by laughedelic**
Here I replace CI script and `deploy` block with [Build Stages](https://docs.travis-ci.com/user/build-stages/) (a beta feature of Travis). It allows you to run release stage only if all jobs from the test stage have passed (without stages `deploy` block is executed in every job, so you need to workaround it with complicated conditions).

I also show here how different test-suites can be run in parallel. It's probably not that useful now, so I can remove it, but it could be useful when we have more time-consuming tests.

Also note that `TEST` env var is used just to distinguish jobs:

screen shot 2017-11-24 at 05 18 54

I hope this also fixes the problem with releasing on merge (and not on normal push). We'll see when we merge this PR :crossed_fingers:

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,37 +1,39 @@
...
+    if: (branch = master) AND (type = push)
+
+jobs:
+  include:
+    # default stage is test
+    - env: TEST="scalafmt"
+      script: ./bin/scalafmt --test
+    - env: TEST="languageserver/test"
+      script: sbt $TEST
+    - env: TEST="tests.ctags"
```
Comment (olafurpg) [inline .travis.yml]: I don't think we need to split the tests into different matrices yet. In #51 I added tests `tests.compiler`, which would not be caught by this. Full test suite still runs in <2-3 minutes.

Comment (olafurpg) [reply .travis.yml]: I suspect we'll want this later down the road however once we get heavier integration tests 👍

Comment (laughedelic) [reply .travis.yml]: OK. Addressed in cb64f24

**Comment by laughedelic** [comment]
> scalameta has just released version fda00d4c of the package metaserver!
Grab it from Bintray now:
https://bintray.com/scalameta/maven/metaserver/fda00d4c
>
>Cheers,
Bintray 

And I just checked that it doesn't release on normal push to master: https://travis-ci.org/scalameta/language-server/builds/306921020 🎉 Finally it works!

**Comment by olafurpg** [comment]
Sweet! Build stages looks very promising, we have had a lot of problems with this in the scalameta build where we publish over 70 artifacts!

**Pull request merged**